tcms/static/js/utils.js

```diff
@@ -100,6 +100,19 @@ function update_component_select_from_product() {
     }
 }
 
+function updateTestPlanSelectFromProduct() {
+    const updateCallback = (data) => {
+        updateSelect(data, '#id_test_plan', 'plan_id', 'name')
+    };
+
+    const productId = $('#id_product').val();
+    if (productId) {
+        jsonRPC('TestPlan.filter', {product: productId}, updateCallback);
+    } else {
+        updateCallback([]);
+    }
+}
```

Comment:
Very similar method in `testing/breakdown.js`. You can refactor the existing method and reuse it (filter the select based on values and pass a call back for the rest). And you can move it to the main utils.js or create a `telemetry/js/utils.js`

Reply:
Also Test Plan is not filtered when Version is changed.